docs/source/userguide/mesh.rst

```diff
@@ -312,3 +312,250 @@ Meshes from FEniCS
+
+To be used with FESTIM, it is necessary for us to define surface and volume markers. 
+
+If the element has been defined explicitly, this is as easy as doing the following:
+```
+id_number = 1
+gmsh.model.addPhysicalGroup(2, [base_surface, cylinder_surface], id_number, name = "surface")
+```
+where the 2 indicates that this is a 2nd dimension element, and we have listed the surfaces that we would like to assign with this ID number.
+
+However, as we generated the surfaces using an extrusion, it can be complicated to keep track of which element corresponds to what.
+GMSH assigns the surface labels cyclically when performing the extrusion, so these element IDs could be directly extracted using code. However, it may be more straightforward and intuitive to open the GUI as before and analyze the surfaces manually. 
+
+After opening the GUI, again after synchronising and using `gmsh.fltk.run()`, go into 'Tools' then 'Options', and ensure that 'Surfaces' is checked under 'Geometry'.
+This will make the surfaces are visible and selectable in the visualisation.
+
+<img width="1710" alt="Screenshot 2024-09-11 at 14 37 53" src="https://github.com/user-attachments/assets/1700db41-2baf-4497-8941-b316190b4963">
+
+We can then hover our mouse over each surface to see its information. For example, we can see that the front tungsten surface is defined as Plane 7, and borders the volume 1. 
+<img width="418" alt="Screenshot 2024-09-11 at 14 38 29" src="https://github.com/user-attachments/assets/79841964-ecd4-4a57-b3f5-4f060f7850ec">
+
+We can now look at each surface, and assign the necessary IDs.
+
+```
+front_id = 1
+back_id = 2
+left_id = 3
+right_id = 4
+top_id = 5
+bottom_id = 6
+outer_cylinder_surface_id = 7
+inner_cylinder_surface_id = 8
+
+tungsten_id = 1
+cucrzr_id = 2
+
+gmsh.model.addPhysicalGroup(2, [7, 10], front_id, name = "front")
+gmsh.model.addPhysicalGroup(2, [6, 9], back_id, name = "back")
+gmsh.model.addPhysicalGroup(2, [1], left_id, name = "left")
+gmsh.model.addPhysicalGroup(2, [3], right_id, name = "right")
+gmsh.model.addPhysicalGroup(2, [4], top_id, name = "top")
+gmsh.model.addPhysicalGroup(2, [2], bottom_id, name = "bottom")
+gmsh.model.addPhysicalGroup(2, [5], outer_cylinder_surface_id, name = "outer_cylinder")
+gmsh.model.addPhysicalGroup(2, [8], inner_cylinder_surface_id, name = "inner_cylinder")
+
+gmsh.model.addPhysicalGroup(3,[1], tungsten_id, name = "tungsten")
+gmsh.model.addPhysicalGroup(3, [2], cucrzr_id, name = "cucrzr")
+```
+
+The model must then be resynchronized before generating the mesh.
+```
+gmsh.model.occ.synchronize()
+
+gmsh.model.mesh.generate(3)
+```
+The mesh can then be written to a file, and GMSH finalised. 
+```
+gmsh.write("my_mesh.msh")
+gmsh.finalize()
+```
+We have now created our mesh! 
+
+However, for use in FESTIM, our mesh now has to be converted into XDMF files, and the surfaces and volume IDs extracted.
```

Comment:
We could have a subsection heading here. The guide could be divided in two: 1) meshing the geometry 2) converting with meshio